### PR TITLE
Make coverage options configurable via env vars

### PR DESCRIPTION
**Pull request by leandro-lucarella-sociomantic**
Now some coverage options can be set via environment variables:

* `DRT_COVMERGE`: Merge current results with pre-existing coverage reports.
* `DRT_COVSRCPATH`: Set path to where source files are located.
* `DRT_COVDSTPATH`: Set path to where listing files are to be written.

### CODE REVIEW

**Comment by leandro-lucarella-sociomantic** [comment]
This is the tangort part of https://github.com/dlang/druntime/pull/1916. I'm only implementing the env vars because doing the whole `--DRT-` CLI options would be too much work for a runtime that should die soonish.

I think we should wait until the druntime PR is merged before merging this.

File: src/rt/cover.d
Comment (nemanja-boric-sociomantic) [inline src/rt/cover.d:124]: ~why `default_` instead of...~ oh, wait, I know the answer :grumpy:

File: src/rt/cover.d
Comment (nemanja-boric-sociomantic) [inline src/rt/cover.d]:
```
srcpath = getenv("DRT_COVSRCPATH", srcpath);
```
for more robust code?

File: src/rt/cover.d
Comment (nemanja-boric-sociomantic) [inline src/rt/cover.d]: ditto

File: src/rt/cover.d
Comment (leandro-lucarella-sociomantic) [reply src/rt/cover.d]: Damn! Yeah, the whole point of using `default_` ( :stuck_out_tongue_closed_eyes: ) was to use it here!

**Review by mihails-strasuns-sociomantic — APPROVED** [review]

**Comment by mihails-strasuns-sociomantic** [comment]
Looks good but also want to test it

**Comment by leandro-lucarella-sociomantic** [comment]
Upstream (druntime) took a different direction, using the more complicated sub-options parsing that I think it would be too much trouble to introduce in tango. I say we deal with the differences elsewhere (makd, beaver), is not like we don't have any differences yet, so it's not a big deal.

If there are no objections by Monday, I'll merge and release v1.8.0.

**Comment by leandro-lucarella-sociomantic** [comment]
> Looks good but also want to test it

Yeah, we should probably go for an rc actually, so it's easier to test it more widely.

**Comment by mihails-strasuns-sociomantic** [comment]
Sounds reasonable (haven't tested it yet myself)

**Comment by nemanja-boric-sociomantic** [comment]
What's the status with this @leandro-lucarella-sociomantic ? We could merge this and release `-rc1` if the GC production testing proves to be as good as it looks now.

**Comment by mihails-strasuns-sociomantic** [comment]
Installed tango built from this PR and tested with ocean - coverage stopped working completely :X

```
export DRT_COVMERGE=1
export DRT_COVDSTPATH=./lst
# modify Build.mak to always use -cov
make test
# not .lst files anywhere
```

**Comment by leandro-lucarella-sociomantic** [comment]
Damn. Have you tried creating the directory?

**Comment by leandro-lucarella-sociomantic** [comment]
OK, I can confirm even creating the directory it seems like it doesn't work. But using only merge does seems to work. I will look into the dst problem.

**Comment by leandro-lucarella-sociomantic** [comment]
(we don't know if that even worked in the first place I guess :D)

**Comment by leandro-lucarella-sociomantic** [comment]
Ha! It seems a combination between merge+first run. It seems like if merge is used it won't create the files :unamused: 

Try to run once without merge and then with merge. The destination directory must exist though.

**Comment by leandro-lucarella-sociomantic** [comment]
Added a new commit to make merge work without pre-existing reports.

**Comment by leandro-lucarella-sociomantic** [comment]
BTW, druntime already implemented this behaviour!

**Comment by leandro-lucarella-sociomantic** [comment]
ping @mihails-strasuns-sociomantic

**Comment by mihails-strasuns-sociomantic** [comment]
Tested with ocean. Works indeed, with only limitation of requiring target dir to be pre-created.

**Review by mihails-strasuns-sociomantic — APPROVED** [review]

**Pull request merged**
